### PR TITLE
[FIX] web_editor: dropdown overflow, offset and double icon

### PR DESCRIPTION
**Pull request by mano-odoo**
The dropdown overflow their container on firefox:
This was due to width fill-available which isn't set on firefox relying on a `width: 100%` effectively occupies the available space better. (moz-available provides weird result on ff >125).

The dropdown-menu was not wide enough, this PR mimics the full width dropdown-menu using the `width: 100%` inheriting the width of the we-row. 

Sometimes we had a misplacement if popper.js consider the dropdown-menu should be rendered at `data-popper-placement`=`x-start`.

Thus why selecting this attribute to adapt the position accordingly.pdown are position relative by default for the dropdown-menu to refer to the position of the dropdown-toggle

Additionnally the tick icon on our dropdown items were displayed twice, this was due to the ::before pseudo element
from the `webclient.scss` still being displayed in website while we manage it already with an ::after element.

task-3527720
part of 3503975

| Before | After |
| -- | -- | 
| image | image | 
| image | image| 
| image| image | 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/scss/wysiwyg_snippets.scss [hunk 1]
```diff
@@ -530,9 +530,16 @@
                     font-weight: bold;
                     padding: 1px;
                 }
+
+                .link-custom-color-border .dropdown-menu.show {
+                    // Compensate positioning to center the border dropdown menu
+                    left: -$o-we-sidebar-content-padding-base !important;
```
Comment (chgo-odoo) [inline addons/web_editor/static/src/scss/wysiwyg_snippets.scss:536]: Having this rule on the dropdown will create an overflow on the `border-color` colorpicker.

I'd rather expect this rule to be applied to the others colorpicker (text-color and fill-color) as shown on the second screenshot, wdyt?

| Border colorpicker (your PR) | Fill color colorpicker (your PR) |
|--------|--------|
| image | image |

File: addons/web_editor/static/src/scss/wysiwyg_snippets.scss [hunk 2]
```diff
@@ -542,11 +549,17 @@
                             display: none;
                         }
                     }
+
```
Comment (chgo-odoo) [inline addons/web_editor/static/src/scss/wysiwyg_snippets.scss:552]: unnecessary diff (?) 🫣

**Review by chgo-odoo — CHANGES_REQUESTED** [review]
Hello @mano-odoo

Thanks for this fix 💪

I just added a small comment about the position of the `dropdown-menu`, otherwise it looks good to me 🙂

File: addons/web_editor/static/src/scss/wysiwyg_snippets.scss (hunk 1 above)
Comment (mano-odoo) [reply addons/web_editor/static/src/scss/wysiwyg_snippets.scss]: Indeed, thanks. It seems the moz-available was providing weird result on firefox >125, I completely changed the way we calculte the dropdown-menu width to avoid this.

**Review by chgo-odoo — APPROVED** [review]
Hello @mano-odoo, thanks for the changes !

I checked the different scenarios we found and it works fine. The visual result is also the expected one.

LGTM 🙂

**Comment by mano-odoo** [comment]
@qsm-odoo could you have a look?

**Review by qsm-odoo — APPROVED** [review]
Thanks :+1: 

@robodoo rebase-ff r+
